go_test.go

```diff
@@ -143,12 +145,39 @@ func TestGoWithGlobalConfig(t *testing.T) {
 
 	prepareGoProject(newHomeDir, t, false)
 	runGo(ModuleNameJFrogTest, tests.GoBuildName, buildNumber, t, "go", "build", "--build-name="+tests.GoBuildName, "--build-number="+buildNumber, "--module="+ModuleNameJFrogTest)
-
 	assert.NoError(t, os.Chdir(wd))
 
 	cleanGoTest()
 }
 
+func TestGoGetSpecificVersion(t *testing.T) {
+	// Init test and preper Global config
+	initGoTest(t)
+	buildNumber := "1"
+	oldHomeDir, newHomeDir := prepareHomeDir(t)
+	defer os.Setenv(coreutils.HomeDir, oldHomeDir)
+	defer os.RemoveAll(newHomeDir)
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	prepareGoProject("", t, true)
+	// Build and publish a go project
+	runGo("", tests.GoBuildName, buildNumber, t, "go", "build", "--build-name="+tests.GoBuildName, "--build-number="+buildNumber)
```

Comment:
We need to check if this "go build" is indeed required, and if so, explain why in a comment.

Reply:
It is needed. I added a more detailed comment.